lib/terraform/runner/api_params.rb

```diff
@@ -42,6 +43,74 @@ def self.to_cam_parameters(vars)
           to_cam_param(key, value)
         end
       end
+
+      require 'set'
+      TRUE_VALUES = ['T', 't', true, 'true', 'True', 'TRUE'].to_set
+
+      # Normalize variables values, from ManageIQ values to Terraform Runner supported values
+      # @param input_vars       [Hash]  key/value pairs as input variables for the terraform-runner run job.
+      # @param type_constraints [Array] array of type constraints objects, from Terraform Runner
+      #
+      # @return [Array] Array of param objects [{name,value,secured}]
+      def self.to_normalized_cam_parameters(input_vars, type_constraints)
+        require 'json'
+
+        input_vars.map do |k, v|
+          type_constr = type_constraints.find { |e| e['name'] == k.to_s }
```

Comment:
It would be more efficient to build a hash indexed on the name key once, maybe you can return this as a hash from the method above [[here]](https://github.com/ManageIQ/manageiq-providers-embedded_terraform/pull/89/files#diff-d9ae67c3672d928afae2d394a3f872f3c2179e5a8ee51c4ffdd1250d51760a5dR161)
```
(payload['input_vars'] || []).index_by { |v| v['name'] }
```

Then here you can just use `type_constraints[k]`